client/src/components/Clan/RoleManagementModal.tsx

```diff
@@ -0,0 +1,315 @@
+import { Backdrop, Box, Checkbox, FormControlLabel, FormGroup, Modal, Paper, TextField } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import FantasyButton from '../FantasyButton';
+import Text from '../Text';
+import { useAlert } from '../../hooks/useAlert';
+import Server from '../../utils/Server';
+import catchError from '../../utils/catchError';
+
+import { ClanGetRolesResponse, ClanPermission } from '@labrute/core';
+
+interface RoleManagementModalProps {
+  open: boolean;
+  onClose: () => void;
+  clanId: string;
+  role?: ClanGetRolesResponse[number] | null;
+  onRoleCreated?: () => void;
+}
+
+const RoleManagementModal: React.FC<RoleManagementModalProps> = ({
+  open,
+  onClose,
+  clanId,
+  role = null,
+  onRoleCreated,
+}) => {
+  const { t } = useTranslation();
+  const Alert = useAlert();
+
+  const [roleName, setRoleName] = useState('');
+  const [permissions, setPermissions] = useState({
+    canAcceptJoinRequests: false,
+    canRejectJoinRequests: false,
+    canRemoveMembers: false,
+    canSelectWarFighters: false,
+    canPinThreads: false,
+    canUnpinThreads: false,
+    canDeletePosts: false,
+    canDeleteThreads: false,
+    canCreateRoles: false,
+    canChangeRoles: false,
+  });
```

Comment:
Why not simply use a permission array?
```suggestion
  const [permissions, setPermissions] = useState<ClanPermission[]>([]);
```